debbuild

```diff
@@ -344,15 +344,21 @@ sub lsb_detection {
         _("Sorry, I quit.\n");
   }
 
+  # if our OS does not have a version, set it to some "unreleased" value
+  $basever //= "9999";
+
   # rpmbuild expects either integers or strings, and some distros (*cough*Ubuntu*cough*)
   # have versions that get interpreted as strings, which creates comparison problems.
   # This will make sure it's an integer to remain compatible.
   (my $specbasever = $basever) =~ s/\.//;
 
   # Set some legacy globals & the standard generic OS-class globals;
   $specglobals{debdist} = $basecodename;
-  $specglobals{debver} = $specglobals{$baseos}
-    = $specglobals{$baseos.'_version'} = $specbasever;
+  $specglobals{debver} = $specbasever;
+  # the OS-class globals shall be overridable from .debmacros, so only set them if they are not yet
+  $specglobals{$baseos} //= $specbasever;
+  # if the basever does not have a sub-version, it needs to be multiplied by 100
+  $specglobals{$baseos.'_version'} //= $basever =~ /\./ || $basever eq "9999" ? $specbasever : $basever * 100;
```

Comment:
I would have split this up such that figuring out the default value was a different statement than acually setting the hash value.

That said, it's still less dense and better commented then many other parts of this codebase, so I'm not going to worry about it. 😄